src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_FixedStatement.cs

```diff
@@ -250,9 +250,22 @@ public override BoundNode VisitFixedLocalCollectionInitializer(BoundFixedLocalCo
             // that is a bit too far
             // we need to pin the underlying field, and only then take the address.
             initializerExpr = ((BoundAddressOfOperator)initializerExpr).Operand;
- 
+
             TypeSymbol initializerType = initializerExpr.Type;
```

Comment:
>TypeSymbol initializerType = initializerExpr.Type; [](start = 12, length = 50)

It looks like before the break, we were getting the type from the BoundAddressOfOperator, before digging through it. Can we restore the same logic instead of trying to patch the type later?

Reply:
The type would still be wrong. The problem is that we need to actually change the type from pointer to reference, but the language sees a fixed buffer as being of pointer type. We could change the type earlier in codegen, but I think that would be a riskier change.

Reply:
I agree that would be riskier. If we feel it is the better change then my suggestion:

1. Keep the fix in the more constrained form here.
1. File a bug to fix the type up earlier in the compiler in the 15.8 release.

Reply:
> but I think that would be a riskier change.

 Before we can make this call, we should actually understand what would be the change. 

---
In reply to: [183831907](https://github.com/dotnet/roslyn/pull/26351#discussion_r183831907) [](ancestors = 183831907)